stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -162,6 +162,8 @@ end
 diagzero(::Diagonal{T}, i, j) where {T} = zero(T)
 diagzero(D::Diagonal{<:AbstractMatrix{T}}, i, j) where {T} = zeros(T, size(D.diag[i], 1), size(D.diag[j], 2))
 
+Base.isstored(D::Diagonal, i, j) = i == j ? Base.isstored(D.diag,i) : false
```

Comment:
Perhaps this should be `isstored(::LinearAlgebra.Diagonal, i::Int, j::Int)`

Reply:
do you mean typing the `ì` and `j` arguments as `int` ? or putting the method in `Base` instead?

Reply:
Adding the type assertions, as this will resolve the amgituity that we see in the tests